### PR TITLE
LPS-114068 Use new Layout Components in Components in dynamic-data-mapping

### PR DESCRIPTION
**Pull request by edalgrin**
This is ~simple~ a code replacement of the layout classes with the new Clay Layout Component.

@matuzalemsteles, can you take a look at this please?

I've tried to replace the code https://clayui.com/docs/components/layout.html but I'm not sure if the dependencies are correct or how to test it because it's a weird combination of Clay 2.x and 3.x

/cc @jbalsas

### CODE REVIEW

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/js/components/Sidebar/Sidebar.es.js [hunk 1]
```diff
@@ -246,9 +247,9 @@ class Sidebar extends Component {
 				<div class={styles} ref="container">
 					<div class="sidebar sidebar-light">
 						<nav class="component-tbar tbar">
-							<div class="container-fluid">
+							<ClayLayout.ContainerFluid>
```
Comment (matuzalemsteles) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/js/components/Sidebar/Sidebar.es.js:250]: Hey @eduardoallegrini, I'm not sure about this but I see that the default of `ContainerFluid` is `container-fluid container-fluid-max-xl` can this cause any collateral effect for this case?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/LocalizableText/LocalizableText.es.js [hunk 2]
```diff
@@ -105,29 +106,34 @@ const LocalesDropdown = ({
...
 							>
-								<span className="autofit-col autofit-col-expand">
-									<span className="autofit-section">
-										<span className="inline-item inline-item-before">
-											<ClayIcon symbol={icon} />
-										</span>
-										{displayName}
-									</span>
-								</span>
+								<ClayLayout.ContentRow containerElement="span">
```
Comment (matuzalemsteles) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/LocalizableText/LocalizableText.es.js:117]: @eduardoallegrini just one question does this replace the removal of `autofit-row` in `<ClayDropDown.Item />`?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/admin/js/main.es.js [hunk 3]
```diff
@@ -463,7 +462,7 @@ class Form extends Component {
 					/>
 				</LayoutProviderTag>
 
-				<div class="container-fluid-1280">
+				<ClayLayout.ContainerFluid>
```
Comment (matuzalemsteles) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/admin/js/main.es.js:465]: hey @eduardoallegrini, I think we can't use the React component in Metal+JSX here 😞

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/admin/js/main.es.js [hunk 4]
```diff
@@ -599,8 +598,7 @@ class Form extends Component {
 			this.refs.ruleBuilder.showRuleCreation();
 
 			this.hideAddButton();
-		}
-		else {
+		} else {
```
Comment (matuzalemsteles) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/admin/js/main.es.js:601]: Just out of curiosity did we change to this pattern again or was that just an SF error?

**Review by matuzalemsteles — COMMENTED** [review]
Hey @eduardoallegrini, sorry for the delay.

Yeah, I kept a lot of CSS stuff when I was migrating, so as not to change the look they expected. And also because I don't really know how everything was organized... but I left just a few comments with some concerns, but it looks very good. If you want to test the modifications there are a few places you may want to go:

- Form (Content & Data -> Form -> Create new Form)
- App Builder (Global Menu -> Control Panel -> Objects -> New Custom Object)

For the reports, you have to activate a property to make the reports visible inside the Form, but I don't know which one is 😅. The same for Web Content with the new Builder.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/js/components/Sidebar/Sidebar.es.js (hunk 1 above)
Comment (eduardoallegrini) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/js/components/Sidebar/Sidebar.es.js:250]: Yeah I know it's not exactly the same but I think it's hardly probable that someone is going to need a space wider than 1280px, so I prefered to limit it thinking it was an error/missing-class in the first place. If somebody ask for this we can fix it easily

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/LocalizableText/LocalizableText.es.js (hunk 2 above)
Comment (eduardoallegrini) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/LocalizableText/LocalizableText.es.js:117]: Yes, Bryce proposed a different solution in this comment https://github.com/liferay-frontend/liferay-portal/pull/56#issuecomment-649076696 but in this case I think we can use the 
dropdown-item as a wrapper for the autofit-row. It makes everything easier to read and change (I think we may have a cleaning/removing-the-unnecessary-element task in the future)

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/admin/js/main.es.js (hunk 4 above)
Comment (eduardoallegrini) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/admin/js/main.es.js:601]: SF error, I launched "VS Code prettier" after "gradlew sf"

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/js/components/Sidebar/Sidebar.es.js (hunk 1 above)
Comment (matuzalemsteles) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/js/components/Sidebar/Sidebar.es.js:250]: Sure, no problem. Just a concern if it causes any side effects visually.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/LocalizableText/LocalizableText.es.js (hunk 2 above)
Comment (matuzalemsteles) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/LocalizableText/LocalizableText.es.js:117]: Hmm, this is interesting. No problem!

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/admin/js/main.es.js (hunk 4 above)
Comment (matuzalemsteles) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/admin/js/main.es.js:601]: 🙂 np

Comment (wincent) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/admin/js/main.es.js:601]: > SF error, I launched "VS Code prettier" after "gradlew sf"

@eduardoallegrini Pro-Tip™: we have instructions [here](https://github.com/liferay/liferay-npm-tools/blob/master/packages/liferay-npm-scripts/README.md#visual-studio-code) for setting up VS Code to format The Liferay Way™ when you're in a liferay-portal checkout (ie. it doesn't mess with how Prettier works in other projects).

**Comment by wincent** [comment]
Needs a rebase.

**Comment by wincent** [comment]
I updated the title for you (ie. using the text from Jira) so that it describes what this PR does and not simply what the last commit does.

**Comment by jbalsas** [comment]
Hey @matuzalemsteles, @eduardoallegrini, not sure if both of you are waiting on the other or on someone else to unblock this... can someone own this, rebase it and see it through this week? 🙏

**Comment by matuzalemsteles** [comment]
haha, I ended up waiting for @eduardoallegrini 😬, I can take care of that without any problems.

**Comment by matuzalemsteles** [comment]
Sending in #82

**Comment by dsanz** [comment]
**pr-check: PASS** — tested on `bc6508f01c5bef1ca3fc099bf055807591fc7db1`

| Validation | Result |
| --- | --- |
| Source Format | PASS |
| Per-Module Compile | PASS |
| Integration Test Compile | PASS |
| Structural Smoke | PASS |
| Java Unit Tests | PASS |

**Pull request closed without merging**